modules/equativBidAdapter.js

```diff
@@ -0,0 +1,146 @@
+    return {
+      data: converter.toORTB({ bidderRequest, bidRequests }),
+      method: 'POST',
+      url: 'https://ssb-global.smartadserver.com/api/bid?callerId=169'
+    };
+  },
+
+  /**
+   * @param serverResponse
+   * @param bidRequest
+   * @returns {Bid[]}
+   */
+  interpretResponse: (serverResponse, bidRequest) =>
+    converter.fromORTB({
+      request: bidRequest.data,
+      response: serverResponse.body,
+    }),
+
+  /**
+   * @param bidRequest
+   * @returns {boolean}
+   */
+  isBidRequestValid: (bidRequest) => {
+    return !!(
+      deepAccess(bidRequest, 'params.networkId') ||
+      deepAccess(bidRequest, 'ortb2.site.publisher.id') ||
+      deepAccess(bidRequest, 'ortb2.app.publisher.id') ||
+      deepAccess(bidRequest, 'ortb2.dooh.publisher.id')
+    );
+  },
+
+  /**
+   * @param syncOptions
+   * @returns {{type: string, url: string}[]}
+   */
+  getUserSyncs: (syncOptions) => {
+    if (syncOptions.iframeEnabled) {
+      window.addEventListener('message', function handler(event) {
+        if (event.origin === COOKIE_SYNC_ORIGIN && event.data.pid) {
+          const exp = new Date();
+          exp.setTime(Date.now() + 31536000000); // in a year
+          storage.setCookie(PID_COOKIE_NAME, event.data.pid, exp.toUTCString());
+          this.removeEventListener('message', handler);
+        }
+      });
+
+      return [{ type: 'iframe', url: COOKIE_SYNC_URL }];
+    }
+
+    return [];
+  }
+};
+
+export const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300,
+  },
+
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    const { siteId, pageId, formatId } = bidRequest.params;
+
+    delete imp.dt;
+
+    imp.bidfloor = imp.bidfloor || getBidFloor(bidRequest);
+    imp.secure = 1;
+    imp.tagid = bidRequest.adUnitCode;
+
+    if (siteId || pageId || formatId) {
+      const bidder = {};
+
+      if (siteId) {
+        bidder.siteId = siteId;
+      }
+
+      if (pageId) {
+        bidder.pageId = pageId;
+      }
+
+      if (formatId) {
+        bidder.formatId = formatId;
+      }
+
+      mergeDeep(imp, {
+        ext: { bidder },
+      });
+    }
+
+    return imp;
+  },
+
+  request(buildRequest, imps, bidderRequest, context) {
+    const bid = context.bidRequests[0];
+    const req = buildRequest(imps, bidderRequest, context);
+
+    if (deepAccess(bid, 'ortb2.site.publisher')) {
+      deepSetValue(req, 'site.publisher.id', bid.ortb2.site.publisher.id || bid.params.networkId);
+    } else if (deepAccess(bid, 'ortb2.app.publisher')) {
+      deepSetValue(req, 'app.publisher.id', bid.ortb2.app.publisher.id || bid.params.networkId);
+    } else if (deepAccess(bid, 'ortb2.dooh.publisher')) {
+      deepSetValue(req, 'dooh.publisher.id', bid.ortb2.dooh.publisher.id || bid.params.networkId);
+    } else {
+      deepSetValue(req, 'site.publisher.id', bid.params.networkId);
+    }
+
+    const pid = storage.getCookie(PID_COOKIE_NAME);
```

Comment:
why not check local storage?

Reply:
Indeed, we're going to improve in the next update.